src/transport/http2.rs

```diff
@@ -80,16 +77,21 @@ pub struct ClientChannel<In: RpcMessage, Out: RpcMessage>(
 );
 
 impl<In: RpcMessage, Out: RpcMessage> ClientChannel<In, Out> {
-    /// create a client given an uri
+    /// create a client given an uri and the default configuration
     pub fn new(uri: Uri) -> Self {
+        Self::new_with_config(uri, ChannelConfig::default())
+    }
+
+    /// create a client given an uri and a custom configuration
+    pub fn new_with_config(uri: Uri, config: ChannelConfig) -> Self {
         let mut connector = HttpConnector::new();
         connector.set_nodelay(true);
         let client = Client::builder()
             .http2_only(true)
-            .http2_initial_connection_window_size(Some(MAX_FRAME_SIZE))
-            .http2_initial_stream_window_size(Some(MAX_FRAME_SIZE))
-            .http2_max_frame_size(Some(MAX_FRAME_SIZE))
-            .http2_max_send_buf_size(MAX_FRAME_SIZE as usize)
+            .http2_initial_connection_window_size(Some(config.max_frame_size))
+            .http2_initial_stream_window_size(Some(config.max_frame_size))
+            .http2_max_frame_size(Some(config.max_frame_size))
+            .http2_max_send_buf_size(config.max_frame_size as usize)
```

Comment:
serious nitpicking: but you could change this cast into `.try_into().unwrap_or(usize::MAX)` or however you spell that.  Or just `.try_into().unwrap()` possibly, panicking is probably reasonable in that case.

Reply:
I think try_into().unwrap() is unreasonable. If an u32 does not fit into an usize, I want to panic.